### ISSUE TITLE
When onPrompt is called, getRemainingTime is stopped; should onAction also be?

### ISSUE THREAD

**Issue by samesfahani-tuplehealth** [SupremeTechnopriest/react-idle-timer#335]
## Question

I am noticing a behavior where when onPrompt is called, the `getRemainingTime` function from the hook is counting down despite activity from the user. I understand the reasoning for this (to support modal confirmations).

Is it true that the `onAction` event should also still be getting called when `isPrompted() === true`?

My use case is that I do server side pings and refreshes on sessions `onAction`, which are still firing when the prompt is up. Of course, I can check for it in my onAction, but `isPrompted` is returned from the hook, so there is a little bit of a chicken and egg problem.

I can work around it with a ref to `isPrompted`, but it's a little ugly.

## Sandbox 1 -- After timer dips below 90000, onAction still gets called but timer keeps counting down
https://codesandbox.io/s/confirm-prompt-forked-kcendr?file=/src/App.tsx

## Sandbox 2 -- onAction not called when prompt active (ref workaround)
https://codesandbox.io/s/confirm-prompt-forked-zcibqu?file=/src/App.tsx

# Summary:
My question is mostly around if `onAction` should still be getting called when `isPrompted() === true`?
And are there any cleaner workarounds?

**Comment by SupremeTechnopriest** [comment]
Yes its designed that way for people who want to automatically close the prompt on user input.  They can listen to `onAction`.  What you can do is make your `onAction` handler a named function so you can use `isPrompted`.

```js
const { isPrompted } = useIdleTimer({ onAction })

function onAction () {
  if (isPrompted()) return
}
```

Javascript wont complain about being defined after its used because named functions are hoisted. Let me know if that works for you.

**Comment by SupremeTechnopriest** [comment]
You can ignore the linter error, or you can just implement your function inside the config:

```js
const { isPrompted } = useIdleTimer({
  onAction: () => {
    if (isPrompted) return
  }
})
```
